src/test/java/core/basesyntax/strategy/impl/OperationStrategyTest.java

```diff
@@ -0,0 +1,85 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.storage.Storage;
+import core.basesyntax.strategy.OperationHandler;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class OperationStrategyTest {
+    private Map<String, OperationHandler> mapOperationHandler;
+    private FruitTransaction fruitTransaction;
+    private String fruitName;
+    private OperationStrategy strategy;
+    private OperationHandler handler;
+
+    @Before
+    public void setUp() {
+        mapOperationHandler = new HashMap<>();
+        mapOperationHandler.put("b", new BalanceOperationHandler());
+        mapOperationHandler.put("p", new PurchaseOperationHandler());
+        mapOperationHandler.put("r", new ReturnOperationHandler());
+        mapOperationHandler.put("s", new SupplyOperationHandler());
+        fruitTransaction = new FruitTransaction("b", new Fruit("banana"), 88);
+        fruitName = "banana";
+        strategy = new OperationStrategy(mapOperationHandler);
+        handler = strategy.getByOperation(fruitTransaction.getOperation());
+        handler.apply(fruitTransaction);
+    }
+
+    @Test
+    public void balanceOperationHandler_88Banana_Ok() {
+        Integer expected = 88;
+        Integer actual = Storage.storage.get(new Fruit(fruitName));
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void purchaseOperation_100Banana_NotOk() {
+        fruitTransaction.setOperation("p");
+        fruitTransaction.setCount(100);
+        strategy = new OperationStrategy(mapOperationHandler);
+        handler = strategy.getByOperation(fruitTransaction.getOperation());
+        handler.apply(fruitTransaction);
+    }
+
+    @Test
+    public void purchaseOperation_80Banana_Ok() {
+        fruitTransaction.setOperation("p");
+        fruitTransaction.setCount(80);
+        strategy = new OperationStrategy(mapOperationHandler);
+        handler = strategy.getByOperation(fruitTransaction.getOperation());
+        handler.apply(fruitTransaction);
+        Integer expected = 8;
+        Integer actual = Storage.storage.get(new Fruit(fruitName));
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void returnOperation_8Banana_Ok() {
+        fruitTransaction.setOperation("r");
+        fruitTransaction.setCount(8);
+        strategy = new OperationStrategy(mapOperationHandler);
+        handler = strategy.getByOperation(fruitTransaction.getOperation());
+        handler.apply(fruitTransaction);
+        Integer expected = 96;
+        Integer actual = Storage.storage.get(new Fruit(fruitName));
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void supplyOperation_12Banana_Ok() {
```

Comment:
I think it's better not to use numbers in the names of test methods! Better focus on the result of the test!

Reply:
I think that this way our test will be more informative in the title